graphql_compiler/compiler/ir_lowering_match/utils.py

```diff
@@ -222,12 +222,17 @@ def construct_where_filter_predicate(simple_optional_root_info):
 CompoundMatchQuery = namedtuple('CompoundMatchQuery', ('match_queries'))
 
 
-class OptionalTraversalTrie(dict):
-    def __init__(self):
+class OptionalTraversalTrie():
```

Comment:
you have to put `object` between the parens here, otherwise in py2 this becomes an old-style object which is problematic